src/runtime/object.cpp

```diff
@@ -1611,9 +1611,23 @@ extern "C" LEAN_EXPORT object * lean_mk_string_from_bytes(char const * s, size_t
 }
 
 extern "C" LEAN_EXPORT object * lean_mk_string(char const * s) {
+    size_t len = strlen(s);
+    if (validate_utf8((const uint8_t *)s, len)) {
+        return lean_mk_string_from_bytes(s, len);
+    } else {
+        return lean_mk_string_core("", 0, 0);
```

Comment:
Is this really a sensible fallback? I matches the panic value but there is no message. Shouldn't this raise an exception if done as part of I/O?

Reply:
Not really. I was planning to put in a follow-up PR which does something like rust's `String::from_utf8_lossy` with replacement characters. In most cases it's not really a good idea to rely on this for error handling, and instead code should call `validate` separately and throw IO errors etc as appropriate.

Reply:
I think it would be good to elaborate on this plan before going forward. It would also be helpful to mention such relevant plans in the PR description :) .

If we want to get to the point where no-one, especially not Lean APIs, depends on this branch, should it become a hard assertion?

Reply:
I have a revision in the works which addresses many of the review comments. I'll post that and then we can talk about how we want to proceed here.